### DISCUSSION TITLE
BBS02 30A current limit

### DISCUSSION THREAD

**Discussion by dcinnovation-mt** [danielnilsson9/bbs-fw#198] [Q&A]
I uploaded the firmware today and it is so much more fun to ride around at 30A compared to 25A! Thanks Daniel for making this available!

I would like to try the full 33A setting, but so far the configuration tool says that 30A is the limit. Is this because of the issues mentioned about a possible hardware current limit? Or is there a setting that I need to change? Using a BBS02

**Comment by danielnilsson9** [comment]
Hi

Thanks for supporting this project!

Yes the max current limit was set in config tool here:
https://github.com/danielnilsson9/bbs-fw/blob/39c373b208a82aa0c3b66520c33543c8ef9835ba/src/tool/Model/Configuration.cs#L147

The BBS02 has a different ADC resolution from BBSHD and should in theory go above 40A if I remember correctly, not tested though.
Of course you risk destroying something but you probably know that :)

**Comment by dcinnovation-mt** [comment]
Haha yeah I know I'm risking some things, but the way I ride even at 30A
the controller barley gets above 35c. The MOSFETs are rated for 200a, and I
wish there was a phase current measurement so I could know how close I am.
The main thing I want from the increased current is short bursts of high
power for climbing short but steep hills.

I'm thinking that the firmware could actually have a simple algorithm that
integrates the amount of time spent at "over current" and then limit it's
availability to no more than like 1 second at 40A, 4 seconds at 35 and 10
seconds at 30amps in a 60 second time window.

If I were to add these features to the source code, how would I compile the
binary?

Happy to support the project! It's made a huge difference in the hills I
can climb already!

-David

**Comment by dcinnovation-mt** [comment]
Okay so I took at look at the data sheet and the MOSFET silicon can handle
210amps, but the bond wires in the package can only handle 120A! Haha maybe
I should be happy with the settings as they are because 30a at the battery
is about 110amps at the MOSFET. Did you get feedback about bbs02s failing
at 33a?

**Reply by CroqueMr** [reply]
I burned so many stock BBS02 controller when starting in the wrong gear just starting (This is why a moved to a BBSHD) you will destroy it with 33A

**Reply by dcinnovation-mt** [reply]
Was that with the old mosfets or the new ones? 30A so far has been fairly good to me. Doing the math it looks like I probably am pushing about 110 Phase amps, so I am coming close to the bond wire limit. If the controller burns out I plan to add stronger mosfets and then push the current higher. I would love to test it on the bench as well to determine how many amps it takes to saturate the core, but that is probably not something I'll get around to

**Reply by CroqueMr** [reply]
I burned 3x new controller (the version with upgraded 3077 mosfet) with original parameter therefore with limit of 25A.
GL if you want to change the mosfet, the whole controller is filled with glue

**Comment by dcinnovation-mt** [comment]
Were you monitoring temps? I know a lot of them die because of heat

On Thu, 26 Oct 2023 at 16:45, CroqueMr ***@***.***> wrote:

> I burned 3x new controller (the version with upgraded 3077 mosfet) with
> original parameter therefore with limit of 25A.
> GL if you want to change the mosfet, the whole controller is filled with
> glue
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/danielnilsson9/bbs-fw/discussions/198#discussioncomment-7397511>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AU7PFCPPL6EO7DLYDSI2TLLYBLKYNAVCNFSM6AAAAAA6ES6IWSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TGOJXGUYTC>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
--
*David Cherry*, *President*
*DC Innovation, LLC*
Phone: (208) 627-3934
dcinnovationmt.com

"Bringing your inventions full circle"

**Reply by CroqueMr** [reply]
It's absolutely not the temperature that burns them. Each time I burned my controller it was at the start, and not necessarily after a big ride, one time even from the first pedal stroke

**Comment by dcinnovation-mt** [comment]
Okay makes sense. What voltage were you running?

On Thu, 26 Oct 2023 at 16:49, CroqueMr ***@***.***> wrote:

> It's absolutely not the temperature that burns them. Each time I burned my
> controller it was at the start, and not necessarily after a big ride, one
> time even from the first pedal stroke
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/danielnilsson9/bbs-fw/discussions/198#discussioncomment-7397528>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AU7PFCOCHGZIJW4KDFSZCITYBLLFTAVCNFSM6AAAAAA6ES6IWSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TGOJXGUZDQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
--
*David Cherry*, *President*
*DC Innovation, LLC*
Phone: (208) 627-3934
dcinnovationmt.com

"Bringing your inventions full circle"

**Reply by CroqueMr** [reply]
48V

**Comment by dcinnovation-mt** [comment]
interesting! I hope Im not next!

On Thu, 26 Oct 2023 at 16:53, CroqueMr ***@***.***> wrote:

> 48V
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/danielnilsson9/bbs-fw/discussions/198#discussioncomment-7397547>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AU7PFCPTKVXVR6YDJZHFMKTYBLLWZAVCNFSM6AAAAAA6ES6IWSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TGOJXGU2DO>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
--
*David Cherry*, *President*
*DC Innovation, LLC*
Phone: (208) 627-3934
dcinnovationmt.com

"Bringing your inventions full circle"

**Reply by CroqueMr** [reply]
Start your engine SYSTEMATICALLY on a low gear, and you should be fine. Everytime when I burned a controller (BBS02 or BBSHD) I accelerated with a high gear engaged.

**Comment by dcinnovation-mt** [comment]
Right,

I have a 28t lekkie bling ring for the front chainring, and I can go up to
44t in the back, so the motor is usually at a high rpm as I am climbing
steep hills in the woods!

On Sun, 29 Oct 2023 at 07:44, CroqueMr ***@***.***> wrote:

> Start your engine SYSTEMATICALLY on a low gear, and you should be fine.
> Everytime when I burned a controller (BBS02 or BBSHD) I accelerated with
> high gear engaged.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/danielnilsson9/bbs-fw/discussions/198#discussioncomment-7415614>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AU7PFCLOJBR33RS7YSW5G2LYBY6SXAVCNFSM6AAAAAA6ES6IWSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TIMJVGYYTI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
--
*David Cherry*, *President*
*DC Innovation, LLC*
Phone: (208) 627-3934
dcinnovationmt.com

"Bringing your inventions full circle"
